### PR TITLE
[FLINK-20601][docs] Rework PyFlink CLI documentation.

### PR DESCRIPTION
**Pull request by shuiqiangchen**
## What is the purpose of the change

*Add PyFlink job submission section under the Advanced CLI section.*

## Brief change log

- *Added a new section named "Submitting PyFlink Jobs" under Advanced CLI section*
- *Corrected the reference links in datastream_tutorial.md and table_api_tutorial.md*
image

### CODE REVIEW

File: docs/deployment/cli.md [hunk 1]
```diff
@@ -338,4 +338,74 @@ specified in the `config/flink-config.yaml`.
...
+{% endhighlight %}
+
+- Submit a Python Table job on a specific JobManager running on host `<jobmanagerHost>` (adapt the command accordingly):
+{% highlight bash %}
+$ ./bin/flink run \
+      --jobmanager <jobmanagerHost>:8081 \
+      --python examples/python/table/batch/word_count.py
+{% endhighlight %}
+
+- Run a Python Table job using a [YARN cluster in Per-Job Mode]({% link deployment/resource-providers/yarn.md %}#run-a-single-flink-job-on-hadoop-yarn):
```
Comment (XComp) [inline docs/deployment/cli.md:387]:
```suggestion
- Run a Python Table job using a [YARN cluster in Per-Job Mode]({% link deployment/resource-providers/yarn.md %}#per-job-cluster-mode %):
```

File: docs/deployment/cli.md [hunk 2]
```diff
@@ -338,4 +338,74 @@ specified in the `config/flink-config.yaml`.
...
+
+<span class="label label-info">Note</span> When submitting Python job via `flink run`, Flink will run the command "python". Please run the following command to confirm that the python executable in current environment points to a supported Python version of 3.5+.
+{% highlight bash %}
+$ python --version
+# the version printed here must be 3.5+
+{% endhighlight %}
+
+The following commands show different PyFlink job submission use-cases:
+
+- Run a Python Table job:
```
Comment (XComp) [inline docs/deployment/cli.md:354]: Thinking about it: Why do we call it Python Table job? Isn't PyFlink job the correct way of labelling it?
This applies to all the occurrences below as well.

File: docs/deployment/cli.zh.md [hunk 3]
```diff
@@ -337,4 +337,74 @@ specified in the `config/flink-config.yaml`.
...
+      -Dkubernetes.cluster-id=<ClusterId> \
+      -Dtaskmanager.memory.process.size=4096m \
+      -Dkubernetes.taskmanager.cpu=2 \
+      -Dtaskmanager.numberOfTaskSlots=4 \
+      -Dkubernetes.container.image=<PyFlinkImageName> \
+      --pyModule word_count \
+      --pyFiles /opt/flink/examples/python/table/batch/word_count.py
+{% endhighlight %}
+
+To learn more available options, please refer to [Kubernetes]({% link deployment/resource-providers/native_kubernetes.md %})
```
Comment (XComp) [inline docs/deployment/cli.zh.md:407]:
```suggestion
To learn more available options, please refer to [Kubernetes]({% link deployment/resource-providers/native_kubernetes.zh.md %})
```

File: docs/deployment/cli.zh.md [hunk 4]
```diff
@@ -337,4 +337,74 @@ specified in the `config/flink-config.yaml`.
...
+or [YARN]({% link deployment/resource-providers/yarn.md %}) which are described in more detail in the
```
Comment (XComp) [inline docs/deployment/cli.zh.md:408]:
```suggestion
or [YARN]({% link deployment/resource-providers/yarn.zh.md %}) which are described in more detail in the
```

File: docs/deployment/cli.zh.md [hunk 5]
```diff
@@ -337,4 +337,74 @@ specified in the `config/flink-config.yaml`.
...
+{% endhighlight %}
+
+- Submit a Python Table job on a specific JobManager running on host `<jobmanagerHost>` (adapt the command accordingly):
+{% highlight bash %}
+$ ./bin/flink run \
+      --jobmanager <jobmanagerHost>:8081 \
+      --python examples/python/table/batch/word_count.py
+{% endhighlight %}
+
+- Run a Python Table job using a [YARN cluster in Per-Job Mode]({% link deployment/resource-providers/yarn.md %}#run-a-single-flink-job-on-hadoop-yarn):
```
Comment (XComp) [inline docs/deployment/cli.zh.md:386]:
```suggestion
- Run a Python Table job using a [YARN cluster in Per-Job Mode]({% link deployment/resource-providers/yarn.zh.md %}#run-a-single-flink-job-on-hadoop-yarn):
```

**Review by XComp — CHANGES_REQUESTED** [review]
Thanks @shuiqiangchen for working on the PyFlink docs. I added a few remarks. The remarks in the English version also apply to the Chinese version.

File: docs/deployment/cli.md (hunk 2 above)
Comment (shuiqiangchen) [reply docs/deployment/cli.md:354]: Yes, Using `PyFlink job` would be more general.

**Comment by shuiqiangchen** [comment]
hi @XComp, thank you for pointing out the mistakes, I have revised them in the latest commit, please have a look.

File: docs/deployment/cli.md [hunk 6]
```diff
@@ -338,4 +338,74 @@ specified in the `config/flink-config.yaml`.
...
+{% endhighlight %}
+
+- Run a PyFlink job with pyFiles:
+{% highlight bash %}
+$ ./bin/flink run \
+      --python examples/python/table/batch/word_count.py \
+      --pyFiles file:///user.txt,hdfs:///$namenode_address/username.txt
+{% endhighlight %}
+
+- Run a PyFlink job with a JAR file:
```
Comment (XComp) [inline docs/deployment/cli.md:366]: Can you come up with reasons here why the user might want to provide a JAR file for his/her PyFlink job?

File: docs/deployment/cli.md [hunk 7]
```diff
@@ -338,4 +338,74 @@ specified in the `config/flink-config.yaml`.
...
+{% highlight bash %}
+$ ./bin/flink run \
+      --python examples/python/table/batch/word_count.py \
+      --jarfile <jarFile>
+{% endhighlight %}
+
+- Run a PyFlink job with pyFiles and pyModule:
```
Comment (XComp) [inline docs/deployment/cli.md:373]: The `--pyModule` parameter is used for specifying the job entry point, isn't it? Could you come up with a more descriptive text here as well?

File: docs/deployment/cli.md [hunk 8]
```diff
@@ -338,4 +338,74 @@ specified in the `config/flink-config.yaml`.
...
+{% endhighlight %}
+
+- Run a PyFlink job using a [YARN cluster in Per-Job Mode]({% link deployment/resource-providers/yarn.md %}#per-job-cluster-mode):
+{% highlight bash %}
+$ ./bin/flink run \
+      --target yarn-per-job
+      --python examples/python/table/batch/word_count.py
+{% endhighlight %}
+
+- Run a PyFlink application on a native Kubernetes cluster having hte cluster ID `<ClusterId>`, it requires a docker image with PyFlink installed, please refer to [Enabling PyFlink in docker]({% link deployment/resource-providers/standalone/docker.md %}#enabling-python):
```
Comment (XComp) [inline docs/deployment/cli.md:394]:
```suggestion
- Run a PyFlink application on a native Kubernetes cluster having the cluster ID `<ClusterId>`, it requires a docker image with PyFlink installed, please refer to [Enabling PyFlink in docker]({% link deployment/resource-providers/standalone/docker.md %}#enabling-python):
```

**Review by XComp — CHANGES_REQUESTED** [review]
Thanks @shuiqiangchen . The changes look good. I just came across an issue with a few of the command descriptions which I missed out in my previous reviews. See my comments below. The comments in the English version apply to the Chinese version as well.

**Comment by XComp** [comment]
@shuiqiangchen I created [FLINK-20601](https://issues.apache.org/jira/browse/FLINK-20601) and assigned it to you. FLINK-20352 is already resolved. Did I select the right Jira user? Please update the PR and commit messages accordingly.

File: docs/deployment/cli.md (hunk 6 above)
Comment (shuiqiangchen) [reply docs/deployment/cli.md:366]: There might be jobs that will reference Java UDF or external connectors. JAR file specified in `--jarfile` will be uploaded to the cluster.

**Comment by shuiqiangchen** [comment]
Hi @XComp, Sorry for the late reply. Thank you for creating the JIRA, I have renamed the commit message and PR title accordingly. Please have a look.

**Review by dianfu — APPROVED** [review]
@shuiqiangchen Thanks a lot for PR and thanks @XComp for the throughout review. +1 from my side.

**Comment by dianfu** [comment]
I'm merging this PR as currently the documentation about how to submit PyFlink jobs is missing. 

@XComp Thanks again for the review.

**Pull request closed without merging**